code/modules/uplink/uplink_item.dm

```diff
@@ -1,21 +1,27 @@
 var/list/uplink_items = list() // Global list so we only initialize this once.
 
+/proc/initialize_global_uplink_items()
+	uplink_items = list()
+	for(var/item in subtypesof(/datum/uplink_item))
+		var/datum/uplink_item/I = new item()
+		if(!I.item)
+			continue
+		if(!uplink_items[I.category])
+			uplink_items[I.category] = list()
+		uplink_items[I.category][I.name] = I
+
 /proc/get_uplink_items(var/datum/game_mode/gamemode = null)
 	if(!uplink_items.len)
-		for(var/item in subtypesof(/datum/uplink_item))
-			var/datum/uplink_item/I = new item()
-			if(!I.item)
-				continue
-			if(!uplink_items[I.category])
-				uplink_items[I.category] = list()
-			uplink_items[I.category][I.name] = I
+		initialize_global_uplink_items()
 
 	var/list/filtered_uplink_items = list()
 	var/list/sale_items = list()
 
 	for(var/category in uplink_items)
 		for(var/item in uplink_items[category])
 			var/datum/uplink_item/I = uplink_items[category][item]
+			if(!I || !istype(I))
```

Comment:
istype is not really necessary here, if something that is not an uplink_item is on this list, then there's something really wrong with it